src/exoscale/deps_modules.clj

```diff
@@ -5,34 +5,50 @@
             [exoscale.deps-modules.path :as p]
             [clojure.spec.alpha :as s]))
 
-(s/def :exo.deps/inherit
-  (s/or :exo.deps.inherit/all #{:all}
-        :exo.deps.inherit/keys (s/coll-of keyword? :min-length 1)))
+(s/def :exoscale.deps/inherit
+  (s/or :exoscale.deps.inherit/all #{:all}
+        :exoscale.deps.inherit/keys (s/coll-of keyword? :min-length 1)))
 
 (set! *warn-on-reflection* true)
 
 (def defaults
   {:dry-run? false
-   :output-deps-edn-file "deps.edn"
-   :input-deps-edn-file "deps.edn"
-   :versions-edn-file ".deps-versions.edn"
-   :versions-edn-keypath []
-   :modules-deps-edn-patterns ["modules/*/deps.edn"]})
+   :versions-file "deps.edn"
+   :versions-keypath [:exoscale.deps/managed-dependencies]
+   :project-files-patterns-keypath [:exoscale.deps/project-files-patterns]})
 
-(defn- find-modules-deps
-  [{:keys [modules-dir modules-deps-edn-patterns input-deps-edn-file]}]
-  (->> (if (some? modules-dir)
-         (p/list-modules-files modules-dir input-deps-edn-file)
-         (p/glob-modules-files "." modules-deps-edn-patterns))))
+(def default-project-files-patterns
+  ["modules/*/deps.edn"])
 
-(defn- deps-edn-out-file
-  [dot-deps-edn-file {:keys [output-deps-edn-file]}]
-  (str (p/sibling dot-deps-edn-file output-deps-edn-file)))
+(defn- find-project-files
+  [{:keys [versions-file project-files-patterns-keypath project-files-patterns
+           modules-dir input-deps-edn-file]}]
+  (let [patterns-from-versions-file (some-> versions-file
+                                            slurp
+                                            edn/read-string
+                                            (get-in project-files-patterns-keypath))]
+    (cond
+      ;; If a `:modules-patterns` config option was given
+      ;; on the command line, let it take precedence
+      (some? project-files-patterns)
+      (p/glob-modules-files "." project-files-patterns)
+
+      ;; If patterns were found in the versions file, let them take
+      ;; precedence
+      (some? patterns-from-versions-file)
+      (p/glob-modules-files "." patterns-from-versions-file)
+
+      ;; If the legacy `modules-dir` key was supplied, use it
```

Comment:
Since the naming changes, I'm not sure it's worth keeping support for this around, wdyt?

Reply:
about "modules-patterns" the naming makes me uneasy. I think the fact it a "pattern" is secondary to what it represent, "a pattern of modules" doesn't say much. I think I prefered "modules-deps-edn-patterns" or "modules-deps-edn-paths" even (it supports wildcards, but it's just paths ultimately).